services/server/src/server/controllers/verification/vyper/stateless/vyper.stateless.handlers.ts

```diff
@@ -0,0 +1,67 @@
+import { Response } from "express";
+import { LegacyVerifyRequest, extractFiles } from "../../verification.common";
+import {
+  VyperCheckedContract,
+  IVyperCompiler,
+  StringMap,
+} from "@ethereum-sourcify/lib-sourcify";
+import { BadRequestError, NotFoundError } from "../../../../../common/errors";
+import { StatusCodes } from "http-status-codes";
+import { getMatchStatus, getResponseMatchFromMatch } from "../../../../common";
+import logger from "../../../../../common/logger";
+import { Services } from "../../../../services/services";
+import { ChainRepository } from "../../../../../sourcify-chain-repository";
+
+export async function verifyVyper(
+  req: LegacyVerifyRequest,
+  res: Response,
+): Promise<any> {
+  const services = req.app.get("services") as Services;
+  const vyperCompiler = req.app.get("vyper") as IVyperCompiler;
+  const chainRepository = req.app.get("chainRepository") as ChainRepository;
+
+  const inputFiles = extractFiles(req);
+  if (!inputFiles) {
+    const msg =
+      "Couldn't extract files from the request. Please make sure you have added files";
+    throw new NotFoundError(msg);
+  }
+
+  const compilerVersion = req.body.compilerVersion; // "0.8.4+commit.c7e474f2";
+  const compilerSettings = req.body.compilerSettings;
+
+  const contractPath = req.body.contractPath;
+  const contractName = req.body.contractName;
```

Comment:
I also saw that, but I was scared of finding a programmatic way to always get rid of the suffix... We always had problems dealing with filepaths and filenames. Do you think we can safely assume to remove everything after the last dot?

Reply:
I think we need to check if the Vyper compiler accepts any arbitrary file suffix or only `.vy` files.

I am just afraid here that the contractName is just accepted from the user as input and never checked against the compiler output. So the user could bring any arbitrary name into the database.

Reply:
I was wrong about my assumption here. The name is actually used to address the output of the recompilation, meaning that a wrong name would lead to an error. 

So it's fine to keep it as it is.